### PR TITLE
Hcap 253 multiple regions fix

### PR DESCRIPTION
**Pull request by mauriciofreshw**
Depends on #165

### CODE REVIEW

**Review by Zenen-freshworks — APPROVED** [review]

File: server/server.js [hunk 1]
```diff
@@ -153,7 +153,14 @@ app.post(`${apiBaseUrl}/approve-user`,
   keycloak.allowRolesMiddleware('ministry_of_health'),
   asyncMiddleware(async (req, res) => {
     await validate(AccessRequestApproval, req.body);
-    await keycloak.approvePendingRequest(req.body.userId, req.body.role, [req.body.region]);
+    const promises = req.body.sites.map((siteId) => dbClient.db[collections.EMPLOYER_SITES].findDoc({ siteId }));
```
Comment (ashtonmeuser) [inline server/server.js:156]: Is the only way to do this via multiple `findDoc` calls? There's no way to query by array?

**Review by fwkendall — DISMISSED** [review]

File: server/server.js [hunk 2]
```diff
@@ -153,7 +153,14 @@ app.post(`${apiBaseUrl}/approve-user`,
...
+    const results = await Promise.all(promises);
+    const siteRegions = results.map(item => item[0].healthAuthority);
```
Comment (ashtonmeuser) [inline server/server.js:158]: Could avoid using the misleading array index `[0]` by using the `single` option of massive `findDoc(query, { single: true }`.

File: server/server.js [hunk 3]
```diff
@@ -153,7 +153,14 @@ app.post(`${apiBaseUrl}/approve-user`,
...
+    const userRegions = siteRegions.includes(req.body.region) ?
+      siteRegions
+      :
+      [...siteRegions, req.body.region];
```
Comment (ashtonmeuser) [inline server/server.js:162]: Okay I see what we're doing here. This is essentially a dedupe for `siteRegions` + `req.body.region`. Alternative here:
```
const joinedRegions = [...siteRegions, req.body.region];
const uniqueRegions = [...new Set(joinedRegions)];
```

**Review by ashtonmeuser — APPROVED** [review]

File: server/server.js (hunk 3 above)
Comment (fwkendall) [reply server/server.js:162]: Yeah this is what I had to state at for a little bit. Adding everything then casting to set is how I would usually approach this as well, if we both find that more readable maybe it's worth swapping to it

**Review by fwkendall — APPROVED** [review]

File: server/server.js [hunk 4]
```diff
@@ -153,7 +153,13 @@ app.post(`${apiBaseUrl}/approve-user`,
   keycloak.allowRolesMiddleware('ministry_of_health'),
   asyncMiddleware(async (req, res) => {
     await validate(AccessRequestApproval, req.body);
-    await keycloak.approvePendingRequest(req.body.userId, req.body.role, [req.body.region]);
+    const results = await dbClient.db[collections.EMPLOYER_SITES].findDoc({
+      or: req.body.sites.map((siteId) => ({ 'siteId ilike': `%${siteId}%` })),
```
Comment (ashtonmeuser) [inline server/server.js:157]: We're using `ilike` quite a bit throughout the app. Shouldn't this be a strict match on something like IDs.

**Review by ashtonmeuser — DISMISSED** [review]

**Review by fwkendall — APPROVED** [review]

**Review by ashtonmeuser — APPROVED** [review]

**Pull request merged**